api/src/event.ts

```diff
@@ -13,13 +13,13 @@ export type EventCallback<T> = (event: Event<T>) => void
  * @param event the event name
  * @param handler the event handler callback
  */
-function listen<T>(
+async function listen<T>(
   event: string,
   handler: EventCallback<T>,
   once = false
-): void {
-  invoke({
-    module: 'Event',
+): Promise<void> {
+  await await invoke({
```

Comment:
Why are there two `await`s here? You could just return the `invoke()` call instead, that way it will return any value that `invoke()` returns, and I think that will help with error handling too.

Reply:
linter bug probably lol